jinahub/indexers/query/vector/NumpyIndexer/__init__.py

```diff
@@ -19,26 +18,36 @@
 
 
 class NumpyIndexer(Executor):
-    def __init__(self, source_path: str, **kwargs):
+    def __init__(self, dump_path: str = None, default_top_k: int = 5, **kwargs):
         super().__init__(**kwargs)
-        ids, vecs = import_vectors(source_path, str(self.metas.pea_id))
-        self._ids = np.array(list(ids))
-        self._vecs = np.array(list(vecs))
-        self._ids_to_idx = {}
+        self.logger = JinaLogger(self.metas.name)
+        self.default_top_k = default_top_k
+        if dump_path is not None:
+            self.logger.info(f'Importing data from {dump_path}')
+            ids, vecs = import_vectors(dump_path, str(self.runtime_args.pea_id))
+            self._ids = np.array(list(ids))
+            self._vecs = np.array(list(vecs))
+            self._ids_to_idx = {}
+        else:
+            self.logger.warning(
+                'No data loaded in "NumpyIndexer". Use .rolling_update() to re-initialize it...'
+            )
 
     @requests(on='/search')
     def search(self, docs: 'DocumentArray', parameters: Dict = None, **kwargs):
         if not self._vecs.size:
             return
 
         if parameters is None:
```

Comment:
Can this ever be the case? I thought parameters is always at least an empty dict.